Add Thermal Resistance (KelvinPerWatt and DegreesCelsiusPerWatt)

andrewmeservy opened the pull request:
- Add Thermal Resistance with KelvinPerWatt and DegreesCelsiusPerWatt

Related discussion: https://github.com/angularsen/UnitsNet/discussions/1558

angularsen commented:
@andrewmeservy Sorry for taking so long, vacations and hectic work life running up to it.
I fixed merge conflicts and removed the old lingering enum values that no longer make sense after the rename.

LGTM, merging.

andrewmeservy commented:
> @andrewmeservy Sorry for taking so long, vacations and hectic work life running up to it. I fixed merge conflicts and removed the old lingering enum values that no longer make sense after the rename.
> 
> LGTM, merging.

no worries at all about the delay, thank you so much for actively maintaining this project, I'm sure it can be a lot of work!! cheers
